components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2146,39 +2146,83 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    async getUsageLimitForTeam(ctx: TraceContext, teamId: string): Promise<number | undefined> {
-        const user = this.checkAndBlockUser("getUsageLimitForTeam");
-        const team = await this.guardTeamOperation(teamId, "get");
-        await this.ensureStripeApiIsAllowed({ team });
+    async getUsageLimit(ctx: TraceContext, attributionId: string): Promise<number | undefined> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
 
-        const attributionId: AttributionId = { kind: "team", teamId };
-        await this.guardCostCenterAccess(ctx, user.id, attributionId, "get");
+        const user = this.checkAndBlockUser("getUsageLimit");
+        switch (attrId.kind) {
+            case "team":
+                const team = await this.guardTeamOperation(attrId.teamId, "get");
+                await this.ensureStripeApiIsAllowed({ team });
+                break;
+            case "user":
+                await this.ensureStripeApiIsAllowed({ user });
+                break;
+        }
+        await this.guardCostCenterAccess(ctx, user.id, attrId, "get");
 
-        const costCenter = await this.usageServiceClientProvider.getDefault().getCostCenter(attributionId);
+        const costCenter = await this.usageServiceClientProvider.getDefault().getCostCenter(attrId);
         if (costCenter) {
             return costCenter.spendingLimit;
         }
         return undefined;
     }
 
-    async setUsageLimitForTeam(ctx: TraceContext, teamId: string, usageLimit: number): Promise<void> {
-        const user = this.checkAndBlockUser("setUsageLimitForTeam");
-        const team = await this.guardTeamOperation(teamId, "update");
-        await this.ensureStripeApiIsAllowed({ team });
+    async setUsageLimit(ctx: TraceContext, attributionId: string, usageLimit: number): Promise<void> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
         if (typeof usageLimit !== "number" || usageLimit < 0) {
-            throw new ResponseError(ErrorCodes.BAD_REQUEST, "Unexpected `usageLimit` value.");
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Unexpected usageLimit value: ${usageLimit}`);
+        }
+
+        const user = this.checkAndBlockUser("setUsageLimit");
+        switch (attrId.kind) {
+            case "team":
+                const team = await this.guardTeamOperation(attrId.teamId, "update");
+                await this.ensureStripeApiIsAllowed({ team });
```

Comment:
This checks for `m.mode === "usage-based" || (m.mode === "chargebee" && !!m.canUpgradeToUBB)` which seems like I could then change my spendinglimit although I haven't put my credit card into stripe, yet. 
Maybe we should reconsider the special 'chargebee' case? What was the idea of it? cc @geropl

Reply:
Let's remove the ènsureStripeApiISAllowed` checks here as well. The check below testing the `billingStrategy !== "stripe"` is doing what we need and is more explicit.

Reply:
@svenefftinge Originally, Chargebee and Usage-Limit was strictly independent. After last weeks discussion, they are not anymore.

> Let's remove the ensureStripeApiIsAllowed checks here as well.

:100: Agreed!

Reply:
@andrew-farries do you agree to remove the calls to `ensureStripeApiIsAllowed`?

Reply:
Yes; I've removed them in a650ae74308b32c2ff910a0e8a1bd3354b990a38